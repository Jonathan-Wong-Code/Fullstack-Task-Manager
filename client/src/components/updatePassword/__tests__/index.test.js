import React from "react";
import UpdatePassword from "..";
import { render, fireEvent, wait, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import axios from "axios";

jest.mock("axios");
afterEach(cleanup);

describe("<updatePassword />", () => {
  test("it should render", () => {
    render(<UpdatePassword />);
  });

  test("should display an error when fields are missing", async () => {
    const error = new Error();
    error.response = { data: { message: "error" } };
    axios.mockImplementation(() => Promise.reject(error));

    const { getByTestId, queryByTestId, container } = render(
      <UpdatePassword />
    );

    expect(queryByTestId("update-pass-message")).toBeNull();
    fireEvent.submit(getByTestId("update-pass-form"));
    await wait(() => {
      expect(getByTestId("update-pass-message")).toBeTruthy();
      expect(getByTestId("update-pass-message")).toHaveTextContent("error");
      expect(container).toMatchSnapshot();
    });
  });
});
