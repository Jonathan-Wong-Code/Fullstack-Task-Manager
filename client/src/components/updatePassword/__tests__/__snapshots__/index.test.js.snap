// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<updatePassword /> should display an error when fields are missing 1`] = `
<div>
  <div>
    <h2>
      Update password
    </h2>
    <form
      action=""
      data-testid="update-pass-form"
    >
      <label
        for="update-password"
      >
        Enter Current Password
      </label>
      <input
        id="update-password"
        type="text"
        value=""
      />
      <label
        for="update-new-password"
      >
        Enter New Password
      </label>
      <input
        id="update-new-password"
        type="text"
        value=""
      />
      <label
        for="update-confirm-updated-password"
      >
        Confirm Updated Password
      </label>
      <input
        id="update-confirm-updated-password"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    
  </div>
</div>
`;

exports[`<updatePassword /> should display an error when fields are missing 2`] = `
<div>
  <div>
    <h2>
      Update password
    </h2>
    <form
      action=""
      data-testid="update-pass-form"
    >
      <label
        for="update-password"
      >
        Enter Current Password
      </label>
      <input
        id="update-password"
        type="text"
        value=""
      />
      <label
        for="update-new-password"
      >
        Enter New Password
      </label>
      <input
        id="update-new-password"
        type="text"
        value=""
      />
      <label
        for="update-confirm-updated-password"
      >
        Confirm Updated Password
      </label>
      <input
        id="update-confirm-updated-password"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <p
      data-testid="update-pass-message"
    >
      error
    </p>
  </div>
</div>
`;
